Firestore/core/src/firebase/firestore/immutable/llrb_node.h

```diff
@@ -302,8 +299,7 @@ LlrbNode<K, V> LlrbNode<K, V>::InnerErase(const K& key,
 
   LlrbNode n = Clone();
 
-  bool descending = comparator(key, n.key());
-  if (descending) {
+  if (util::Ascending(comparator.Compare(key, n.key()))) {
```

Comment:
You've got this inversed from previously... sort of. Previously, line 303 was for the *descending* case, and now it's for the *ascending* case. But actually, I think everything's ok, and the previous code was misleading, though still correct. (i.e. it should've been `bool *ascending* = comparator(key, n.key());`). But better double check that.

Reply:
Thankfully we have great tests for this because I had originally done this the other way around, getting confused just the way you suggested. The tests and the Android port agree though, that this is the right way.